### PR TITLE
Attempt at creating services example that includes rendered output

### PR DESCRIPTION
**Pull request by SimonBiggs**
I wanted to be able to create an example that includes rendered output. Here is my attempt. Unfortunately both this and the base browser example has the following typescript error:

```
$ tsc && webpack
node_modules/@jupyterlab/coreutils/lib/interfaces.d.ts(21,38): error TS1005: ',' expected.
node_modules/@jupyterlab/coreutils/lib/interfaces.d.ts(21,40): error TS2300: Duplicate identifier 'T'.
node_modules/@jupyterlab/coreutils/lib/statedb.d.ts(23,35): error TS2314: Generic type 'IDataConnector<T, U>' requires 2 type argument(s).
error Command failed with exit code 1.
```

Nevertheless, the code I have written here is a reasonable base for what an example achieving this would look like.

I am keen for feedback.

Cheers,
Simon

### CODE REVIEW

File: packages/services/examples/typescript-browser-with-output/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "es6-promise": "^4.1.0"
+  },
+  "scripts": {
+    "build": "tsc && webpack",
+    "clean": "rimraf build"
+  },
+  "devDependencies": {
+    "rimraf": "^2.5.2",
+    "tslint": "~4.5.0",
+    "typescript": "~2.2.0",
```
Comment (ian-r-rose) [inline packages/services/examples/typescript-browser-with-output/package.json]: I believe cause of the error you were running into is here. We updated to TS 2.4.1 a day or two ago, and having and older version causes some type errors.

Comment (SimonBiggs) [inline packages/services/examples/typescript-browser-with-output/package.json]: Thanks Ian,

That has successfully changed the error :). That's positive. Upon locking to typescript 2.4.1 I now get the following:

```
$ yarn build
yarn build v0.27.5
$ tsc && webpack
node_modules/@jupyterlab/services/node_modules/@jupyterlab/coreutils/lib/modeldb.d.ts(265,22): error TS2420: Class 'ModelDB' incorrectly implements interface 'IModelDB'.
  Types of property 'createList' are incompatible.
    Type '(path: string) => IObservableUndoableList<JSONValue>' is not assignable to type '<T extends JSONValue>(path: string) => IObservableUndoableList<T>'.
      Type 'IObservableUndoableList<JSONValue>' is not assignable to type 'IObservableUndoableList<T>'.
        Type 'JSONValue' is not assignable to type 'T'.
          Type 'string' is not assignable to type 'T'.

```

**Comment by SimonBiggs** [comment]
@ian-r-rose just wanted to let you know that you don't need to prioritise fixing this for me. I have this code working successfully in my own code base.

I just figured I may as well write an example so that other newcomers are able to achieve the same.

File: packages/services/examples/typescript-browser-with-output/package.json (hunk 1 above)
Comment (ian-r-rose) [inline packages/services/examples/typescript-browser-with-output/package.json]: Looks to me like you are getting an old version of `@jupyterlab/coreutils` (from a whole 36 hours ago) somewhere in your `node_modules`.

**Comment by ian-r-rose** [comment]
No worries, @SimonBiggs , I went through the same process earlier today.

**Comment by SimonBiggs** [comment]
Thanks for all the help @ian-r-rose :). It worked. So this is a just a simple example that runs some code within the kernel and renders the output in the browser.

Hopefully it helps someone. :).

**Comment by SimonBiggs** [comment]
Give us a moment to update the readme.

**Comment by SimonBiggs** [comment]
Done :).

**Comment by blink1073** [comment]
Thanks for this, @SimonBiggs!  Would you mind if we removed the use of `yarn` to conform with the rest of the examples?  We may switch the whole repo to use `yarn`, but we haven't gotten there yet.

**Comment by SimonBiggs** [comment]
Sure can. I'll have to do that tomorrow.

**Comment by SimonBiggs** [comment]
Or you can feel free to do it.

**Comment by SimonBiggs** [comment]
Hi Steven @blink1073,

I've just swapped out yarn for npm and updated the readme. It's been a pleasure writing up my first jupyter project pull request :).

**Comment by SimonBiggs** [comment]
Would there be any value in submitting an example that uses Angular. A reactive form making use of the Jupyter services?

**Comment by blink1073** [comment]
I submitted a PR against your fork (https://github.com/SimonBiggs/jupyterlab/pull/1) with a bit of cleanup.  I have not personally used Angular, so I am less inclined to add an example.

**Comment by SimonBiggs** [comment]
I'm offering to write the angular example :). I've essentially already written the code in my own project. But I could pull out the key parts and throw them into an example.

**Comment by SimonBiggs** [comment]
Downside of an angular example is that there is a lot of boilerplate code. The boilerplate code comes from using the Angular CLI, but it would be overwhelming for someone trying to read the code who hasn't seen am Angular code base before.

There might be value in having an example, but it might also not be very helpful.

**Comment by SimonBiggs** [comment]
Another part of my motivation of wanting to write examples is I am licensing my own code base under the AGPL (https://github.com/SimonBiggs/scriptedforms/releases/tag/v0.1.0-alpha.2). If in the process of writing I work something out that other people using Jupyter tools can make use of I would like them to be able to copy that under the same terms as the Jupyter project. But given my industry is small I want something like the AGPL in my own code base to guarantee that the version most people are using is one that is transparent and available.

Feel kind of bad taking so much of your work and then licensing my work built on top of it in such a way that code can't be pulled back.

**Comment by blink1073** [comment]
Specifically, my concern is in maintaining the example, since none of the core maintainers use Angular.

**Comment by blink1073** [comment]
This looks great, thanks!

**Comment by SimonBiggs** [comment]
Makes sense about the angular example. Maybe what I'll do is just make my own repo with a collection of examples under an Apache licence.

**Comment by SimonBiggs** [comment]
And you're welcome. My pleasure. Thank you for building all the pieces in such a modular way that I can take it and build something completely different with ease standing on the shoulders of the work you've already done.

**Comment by blink1073** [comment]
Cheers!

**Pull request merged**